pkg/resources/openstack/lb.go

```diff
@@ -17,43 +17,93 @@ limitations under the License.
 package openstack
 
 import (
-	"fmt"
+	"strings"
 
+	"github.com/gophercloud/gophercloud/openstack/loadbalancer/v2/listeners"
 	"github.com/gophercloud/gophercloud/openstack/loadbalancer/v2/loadbalancers"
+	v2pools "github.com/gophercloud/gophercloud/openstack/loadbalancer/v2/pools"
 	"k8s.io/kops/pkg/resources"
 	"k8s.io/kops/upup/pkg/fi"
 	"k8s.io/kops/upup/pkg/fi/cloudup/openstack"
 )
 
 const (
-	typeLB = "lb"
+	typeLB  = "LoadBalancer"
+	typeLBL = "LBListener"
+	typeLBP = "LBPool"
 )
 
-func listLBs(cloud openstack.OpenstackCloud, clusterName string) ([]*resources.Resource, error) {
+func (os *clusterDiscoveryOS) ListLB() ([]*resources.Resource, error) {
+	var resourceTrackers []*resources.Resource
 	opts := loadbalancers.ListOpts{
-		Name: clusterName,
+		Name: "api." + os.clusterName,
```

Comment:
It would be desirable to eventually create a single source of truth for names like this

Reply:
That's fair - and if you're able to use tags/metadata instead that's even better than relying on the name!  Some of these are just limitations of AWS, or historical limitations that we have to accommodate still.